backend/src/main/java/com/bakdata/conquery/io/mina/MdcFilter.java

```diff
@@ -0,0 +1,47 @@
+package com.bakdata.conquery.io.mina;
+
+import com.bakdata.conquery.util.io.ConqueryMDC;
+import lombok.RequiredArgsConstructor;
+import org.apache.mina.core.filterchain.IoFilterEvent;
+import org.apache.mina.filter.util.CommonEventFilter;
+
+@RequiredArgsConstructor
+public class MdcFilter extends CommonEventFilter {
+
+	private final ThreadLocal<Integer> callDepth = ThreadLocal.withInitial(() -> 0);
+
+	private final String locationFmt;
+
+	/**
+	 * Adapted from {@link org.apache.mina.filter.logging.MdcInjectionFilter}
+	 */
+	@Override
+	protected void filter(IoFilterEvent event) throws Exception {
+
+		// since this method can potentially call into itself
+		// we need to check the call depth before clearing the MDC
+		int currentCallDepth = callDepth.get();
+		callDepth.set(currentCallDepth + 1);
```

Comment:
@awildturtok in dem alten PR hattest du hier `getAndIncrement` vorgeschlagen, aber es gibt soweit ich weiß kein ThreadLocal für primitive Datentypen, wie ein ThreadLocalInt

Reply:
ah, ich hatte vermutlich gedacht, dasses atomic ist.